client/network/src/protocol/sync/blocks.rs

```diff
@@ -165,20 +164,19 @@ impl<B: BlockT> BlockCollection<B> {
 	pub fn drain(&mut self, from: NumberFor<B>) -> Vec<BlockData<B>> {
 		let mut drained = Vec::new();
 		let mut ranges = Vec::new();
-		{
-			let mut prev = from;
-			for (start, range_data) in &mut self.blocks {
-				match range_data {
-					&mut BlockRangeState::Complete(ref mut blocks) if *start <= prev => {
-							prev = *start + (blocks.len() as u32).into();
-							let mut blocks = mem::replace(blocks, Vec::new());
-							drained.append(&mut blocks);
-							ranges.push(*start);
-					},
-					_ => break,
-				}
+
+		let mut prev = from;
+		for (start, range_data) in &mut self.blocks {
+			match range_data {
+				&mut BlockRangeState::Complete(ref mut blocks) if *start <= prev => {
+						prev = *start + (blocks.len() as u32).into();
+						drained.append(blocks);
```

Comment:
`append` isn't a very descriptive name, and I had to look up the documentation to check whether this was draining the content of `blocks`. Maybe a small comment mentioning that wouldn't hurt.

Reply:
Done.

Reply:
Documentation says `Moves all the elements of other into Self, leaving other empty.` but it does not clarify if "empty" means memory is deallocated.  `mem::replace` makes sure the memory is freed when `blocks` is dropped`

Reply:
Yeah, but this entry is going to be removed from the hashmap anyway. That will drop the vec.